src/build/env.py

```diff
@@ -169,20 +179,26 @@ def _create_isolated_env_venv(path):  # type: (str) -> Tuple[str, str]
     import venv
 
     venv.EnvBuilder(with_pip=True).create(path)
-    executable, script_dir = _find_executable_and_scripts(path)
-    # avoid the setuptools from ensurepip to break the isolation
-    if sys.version_info < (3, 6, 6):  # python 3.5 up to 3.6.5 come with pip 9 that's too old, for new standards
+    executable, script_dir, purelib = _find_executable_and_scripts(path)
+
+    # Require a minimum version of pip (currently 10)
+    (pip_version,) = (
+        Version(d.version) for d in metadata.distributions(path=[purelib]) if canonicalize_name(d.metadata['name']) == 'pip'
```

Comment:
do we need canonicalize_name? I don't think so, pip is same in all formats

Reply:
also instead of iterating through all can't we just lookup pip?

Reply:
Not sure if there's a simple API to do so, lookup by name doesn't have the ability to select a path. Also, this should be a very short list because it's a new venv?

Reply:
Why not use the singular `metadata.distribution(d.metadata['name'], path=[purelib])` instead? There’s also a shortcut `metadata.version(d.metadata['name'], path=[purelib])` if you just want the version, but I didn’t check its availability in the stdlib module.

Reply:
Oops, I took too long to reply and didn’t see the cot above. It seems like the shortcut functions indeed does not take context kwargs.

Reply:
This may still be simpler though:

```python
next(metadata.distributions(name=d.metadata['name'], path=[purelib])).version
```

Also note that it is not safe to unpack the iterator directly into a one-element tuple. It is possible for importlib-metadata to return multiple pip copies if it finds e.g. one in system-site-packages and one in user-site.

Reply:
But we are only passing one path, so isn't multiple pip's an error we want to see?